packages/expo-device/ios/EXDevice/EXDevice.m

```diff
@@ -0,0 +1,256 @@
+{
+#if TARGET_OS_TV
+  BOOL isPinOrFingerprintSet = NO;
+#else
+  LAContext *context = [[LAContext alloc] init];
+  BOOL isPinOrFingerprintSet = ([context canEvaluatePolicy:LAPolicyDeviceOwnerAuthentication error:nil]);
+#endif
+  resolve(@(isPinOrFingerprintSet));
+}
+
+UM_EXPORT_METHOD_AS(getFreeDiskStorageAsync, getFreeDiskStorageAsyncWithResolver:(UMPromiseResolveBlock)resolve rejecter:(UMPromiseRejectBlock)reject)
+{
+  resolve([self freeDiskStorage]);
+}
+
+- (NSString *)deviceId
+{
+  struct utsname systemInfo;
+  
+  uname(&systemInfo);
+  
+  NSString* deviceId = [NSString stringWithCString:systemInfo.machine
+                                          encoding:NSUTF8StringEncoding];
+  
+  if ([deviceId isEqualToString:@"i386"] || [deviceId isEqualToString:@"x86_64"] ) {
+    deviceId = [NSString stringWithFormat:@"%s", getenv("SIMULATOR_MODEL_IDENTIFIER")];
+  }
+  
+  return deviceId;
+}
+
+- (NSString *)carrier
+{
+#if (TARGET_OS_TV)
+  return nil;
+#else
+  CTTelephonyNetworkInfo *netinfo = [[CTTelephonyNetworkInfo alloc] init];
+  CTCarrier *carrier = [netinfo subscriberCellularProvider];
+  return carrier.carrierName;
+#endif
+}
+
+
+- (NSString *)deviceType
+{
+  switch ([[UIDevice currentDevice] userInterfaceIdiom]) {
+    case UIUserInterfaceIdiomPhone: return @"Handset";
+    case UIUserInterfaceIdiomPad: return @"Tablet";
+    case UIUserInterfaceIdiomTV: return @"Tv";
+    default: return @"Unknown";
+  }
+}
+
+- (bool) isTablet
+{
+//  return [[self deviceType] isEqualToString:@"Tablet"];
+  return [[UIDevice currentDevice] userInterfaceIdiom] == UIUserInterfaceIdiomPad;
+}
+
+- (NSString *)getCPUType {
+  /* https://stackoverflow.com/questions/19859388/how-can-i-get-the-ios-device-cpu-architecture-in-runtime */
+  const NXArchInfo *info = NXGetLocalArchInfo();
+  NSString *typeOfCpu = [NSString stringWithUTF8String:info->description];
+  return typeOfCpu;
+}
+
+
+- (NSNumber *)totalMemory {
+  return [NSNumber numberWithUnsignedLongLong:[NSProcessInfo processInfo].physicalMemory];
+}
+
+- (NSDictionary *)documentFileSystemAttributes {
+  NSArray *paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
+  return [[NSFileManager defaultManager] attributesOfFileSystemForPath:[paths lastObject] error:nil];
+}
+
+- (NSNumber *)totalDiskCapacity {
+  NSDictionary *storage = [self documentFileSystemAttributes];
+  
+  if (storage) {
+    NSNumber *fileSystemSizeInBytes = storage[NSFileSystemSize];
+    return fileSystemSizeInBytes;
+  }
+  return nil;
+}
+
+- (NSNumber *)freeDiskStorage {
+  NSDictionary *storage = [self documentFileSystemAttributes];
+  
+  if (storage) {
+    NSNumber *freeFileSystemSizeInBytes = storage[NSFileSystemFreeSize];
+    return freeFileSystemSizeInBytes;
+  }
+  return nil;
+}
+
+
+- (NSDictionary *)constantsToExport
+{
+  UIDevice *currentDevice = [UIDevice currentDevice];
+  NSString *uniqueId = [EXDeviceUID uid]; //TODO: need to import this
```

Comment:
is this todo still relevant? (same question about all of the comments in the dictionary being returned)